contract/contracts/Voting/Types.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+library Types {
+      struct DirectiveVotes {
+        uint256 yes;
+        uint256 no;
+        uint256 abstain;
+    }
+
+    struct Candidate {
+        string name;
+        address candidateAddress;
+        uint256 votes;
+    }
+    struct Member{
+        string name;
+        bool isEligible;
+        bool isVoted;
+        address memberAddress;
+    }   
+    struct Proposal{
+        string title;
+        string description;
+        string draftedBy;
+        bool isElection;
+        bool isActive;
+        DirectiveVotes votes;
+        Candidate[] candidates;
+        Member[] voters;
```

Comment:
for voters i prefer to use mapping(address => boolean), so we dont need to do looping in `findVoter` so it will be constant time complexity O(1)

Reply:
The reason why I have avoided mappings is because **_Assignments to structs or arrays in storage do not work if they contain mappings_**